### PR TITLE
custom-exception

### LINKED ISSUE #2663: Raise a custom exception when Last-Modified is not available
So we don't have to resort to https://github.com/mozilla/bugbug/blob/fa24337fc17c725c3017bc3ad303b4d8e43c7757/bugbug/phabricator.py#L150 but can directly catch the specific exception we want to catch.

### PR DESCRIPTION
**Pull request by anooptp**
Fixes #2663

### CODE REVIEW

File: bugbug/models/exception.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+class CustomException(Exception):
```
Comment (marco-c) [inline bugbug/models/exception.py:7]: Let's define this directly in bugbug/db.py, and let's call it "LastModifiedNotAvailable"

Comment (marco-c) [reply bugbug/models/exception.py:7]: Doing this should also avoid the circular import problem that you're hitting.

Comment (anooptp) [reply bugbug/models/exception.py]: implemented the changes mentioned

File: bugbug/db.py [hunk 2]
```diff
@@ -152,6 +152,14 @@ def read(self):
             pass
 
 
+class LastModifiedNotAvailable(Exception):
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return self.message
```
Comment (marco-c) [inline bugbug/db.py:160]: We can implement this as an empty class so it inherits everything from its parent `Exception`:
```
class LastModifiedNotAvailable(Exception):
    pass
```

File: bugbug/db.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def last_modified(path):
     last_modified = utils.get_last_modified(url)
 
     if last_modified is None:
-        raise Exception("Last-Modified is not available")
+        raise LastModifiedNotAvailable("Last-Modified is not available")
```
Comment (marco-c) [inline bugbug/db.py:122]: The name of the class says it all, we can pass an empty message now:
```suggestion
        raise LastModifiedNotAvailable()
```

File: bugbug/phabricator.py [hunk 4]
```diff
@@ -146,9 +147,9 @@ def download_revisions(rev_ids: Collection[int]) -> None:
 def download_modified_revisions():
     try:
         last_modified = db.last_modified(REVISIONS_DB)
-    except Exception as e:
-        if str(e) == "Last-Modified is not available":
-            return
+    except LastModifiedNotAvailable as e:
+        logger.error("Exception:", e)
```
Comment (marco-c) [inline bugbug/phabricator.py:151]: Let's keep the same behavior as before: no logging

File: bugbug/db.py (hunk 2 above)
Comment (anooptp) [reply bugbug/db.py]: implemented

File: bugbug/phabricator.py (hunk 4 above)
Comment (anooptp) [reply bugbug/phabricator.py]: excluded logging

File: bugbug/db.py [hunk 5]
```diff
@@ -152,6 +152,10 @@ def read(self):
             pass
 
 
+class LastModifiedNotAvailable(Exception):
```
Comment (marco-c) [inline bugbug/db.py:155]: Nit: can you move this to earlier than the `last_modified` function or at the top of the file? I prefer the definitions to be earlier than the usages.

Comment (anooptp) [reply bugbug/db.py]: Moved to top of file.
Had kept it there so that all class definitions are together.

**Review by marco-c — APPROVED** [review]

**Comment by marco-c** [comment]
Thanks!

**Comment by anooptp** [comment]
Thank you!

**Review by webbdays — CHANGES_REQUESTED** [review]
We can just use db.LastModifiedNotAvailable in test_db.py

**Comment by webbdays** [comment]
@anooptp 
@marco-c 
Can I make that change ?

**Comment by marco-c** [comment]
@webbdays sure, you can submit another PR for that. It's a minor cosmetic change though, not so important :)

**Comment by webbdays** [comment]
Ok, I will do this in the current PR #2665

**Comment by webbdays** [comment]
It's merged, I will do it in new PR.

**Pull request merged**